packages/flutter_tools/lib/src/mdns_discovery.dart

```diff
@@ -57,15 +57,27 @@ class MDnsObservatoryDiscovery {
   /// it will return that instance's information regardless of what application
   /// the Observatory instance is for.
   @visibleForTesting
-  Future<MDnsObservatoryDiscoveryResult?> query({String? applicationId, int? deviceVmservicePort}) async {
+  Future<MDnsObservatoryDiscoveryResult?> query({
+    String? applicationId,
+    int? deviceVmservicePort,
+    bool ipv6 = false,
+    bool isNetworkDevice = false
+  }) async {
     _logger.printTrace('Checking for advertised Dart observatories...');
     try {
       await _client.start();
-      final List<PtrResourceRecord> pointerRecords = await _client
-        .lookup<PtrResourceRecord>(
-          ResourceRecordQuery.serverPointer(dartObservatoryName),
-        )
-        .toList();
+
+      List<PtrResourceRecord> pointerRecords = <PtrResourceRecord>[];
+      if (isNetworkDevice) {
+        // Keep checking for records every 2 seconds.
```

Comment:
It looks like this restarts the lookup every 2 seconds.  Can we instead let it look for longer, since the caller will handle "this is taking longer than expected"?

https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#never-check-if-a-port-is-available-before-using-it-never-add-timeouts-and-other-race-conditions

Reply:
How long did you have in mind? The default was 5 seconds, but when I tested it, 2 seconds was always enough time but maybe that's an invalid assumption. And this lookup does not end when it finds it, it will end at the timeout. So if you put a timeout of 30 seconds, it will run for 30 seconds regardless of what it finds. So if we put a higher timeout, that's longer that the user has to wait for it to finish even though it probably already found what it's looking for.

Reply:
> And this lookup does not end when it finds it, it will end at the timeout.

It looks like it's returning a `Stream`, are you sure it's not waiting that long because you added a `toList()` at the end, which _will_ wait until the stream is closed?
https://github.com/flutter/packages/blob/e1ab965985d4e8ab5fdcf1b9989cf1c22d770205/packages/multicast_dns/lib/src/lookup_resolver.dart#L48

See their `await for` example: 
https://github.com/flutter/packages/blob/88585ef98897007eacc1895f90e1288b5624b960/packages/multicast_dns/example/main.dart#L21-L22

Reply:
You're right, I see. I refactored so it stays open for longer (10 minutes). Let me know if you think longer/shorter time would be better